projects/fundamental-ngx/src/lib/utils/custom-class-base-component.ts

```diff
@@ -0,0 +1,33 @@
+import { ElementRef, OnChanges, OnInit, Input } from '@angular/core';
+
+export abstract class CustomClassBaseComponent implements OnInit, OnChanges {
+    private _elementRef: ElementRef;
+
+    @Input() class; // user's custom classes
+
+    abstract _setProperties(): void;
```

Comment:
maybe we should document the purpose of that function and the one after for future